models/interactions.sql

```diff
@@ -0,0 +1,195 @@
+{{ 
+  config(
+    materialized='table',
+    unique_key = 'event_id',
+    sort = 'start_tstamp',
+    dist = 'event_id'
+  )
+}}
+
+with prep as (
+
+ 	select
+    e.event_id,
+    e.page_view_id,
+    e.domain_sessionid,
+    e.domain_userid,
+    coalesce(y.player_id, me.html_id, r.content_title) as media_id, -- replace content_title with id field when available
+    round(mp.duration) as duration,
+    coalesce(y.player_id, me.html_id, r.content_title) as title, -- replace ids with title fields when available
+    case when me.media_type = 'audio' then 'audio' else 'video' end as media_type,
+    coalesce(y.schema_vendor||'-'||y.schema_name, me.schema_vendor||'-'||me.schema_name, r.schema_vendor||'-'||r.schema_name)  as media_player_type,
+    e.page_referrer,
+    coalesce(e.page_url, r.content_url) as content_url,
+    e.geo_region_name,
+    e.br_name,
+    e.dvce_type,
+    e.os_name,
+    e.os_timezone,
+    mpe.type as event_type,
+    e.derived_tstamp as start_tstamp, 
+    mp.current_time as media_start_pos,
+    isnull(mp.playback_rate, 1) as playback_rate,
+    coalesce(y.playback_quality, ve.video_height||'x'||ve.video_width, rd.display_height||'x'||rd.display_width) as playback_quality,
+    case when event_type = 'ended' then 100 else mp.percent_progress end percent_progress,
```

Comment:
It is up to the user to define which % progress they would like to track, however, as there is a separate `ended` event, they might not add `100` to that list as it would be superfluous to have an extra event just for that (the default values are 15, 25, 50 and 75), that is why for now I opted for updating the percent progress field in those cases to 100%. I'm expecting this to be the only 'correction' we would consider making even though there are many different event types.